src/shared/components/cancerSummary/CancerSummary.tsx

```diff
@@ -0,0 +1,448 @@
+    };
+}
+
+interface ICancerSummaryContentProps {
+    data: {
+        [cancerType:string]:ICancerAlterationData
+    };
+}
+
+export interface ICancerSummaryContentState {
+    yAxis: 'alt-freq' | 'abs-count';
+    xAxis: 'y-axis' | 'can-types';
+    altSamples: {value: number; max: number};
+    totalSamples: {value: number; max: number};
+    showGenomicAlt: boolean;
+    cancerType: string;
+    datasets?: {
+        data: IBarGraphDatasets[];
+        label: string;
+        sortCount: number;
+        sortBy: string;
+    }[];
+}
+
+@observer
+export class CancerSummaryContent extends React.Component<ICancerSummaryContentProps, ICancerSummaryContentState> {
+
+
+    private inputYAxisEl:any;
+    private inputXAxisEl:any;
+    @observable private pngAnchor:any = '';
+    @observable private pdfAnchor = '';
+
+    @observable private showControls=false;
+    @observable private showGenomicAlt = true;
+    @observable private yAxis: 'alt-freq' | 'abs-count' = 'alt-freq';
+    @observable private xAxis: 'y-axis' | 'can-types' = 'y-axis';
+    @observable private multiSelectValue:string = this.cancerTypes[0].value;
+    @observable private multiSelect:any = {
+        options: this.cancerTypes,
+        value: this.multiSelectValue
+    };
+    @observable private altCasesValue = 0;
+    @observable private totalCasesValue = 0;
+
+    constructor(props:ICancerSummaryContentProps) {
+        super(props);
+
+        this.handleYAxisChange = this.handleYAxisChange.bind(this);
+        this.handleXAxisChange = this.handleXAxisChange.bind(this);
+        this.handleAltSliderChange = this.handleAltSliderChange.bind(this);
+        this.handleGenomicCheckboxChange = this.handleGenomicCheckboxChange.bind(this);
+        this.handleSelectChange = this.handleSelectChange.bind(this);
+        this.handleTotalSliderChange = this.handleTotalSliderChange.bind(this);
+        this.toggleShowControls = this.toggleShowControls.bind(this);
+        this.setPngAnchor = this.setPngAnchor.bind(this);
+        this.setPdfAnchor = this.setPdfAnchor.bind(this);
+        this.downloadPdf = this.downloadPdf.bind(this);
+        this.resetSliders = this.resetSliders.bind(this);
+    }
+
+    @computed get datasets() {
+        const {data} = this.props;
+        const {yAxis, showGenomicAlt, multiSelectValue:msv} = this;
+        return _.without(
+            _.map(data, (cancer:ICancerAlterationData, cancerTypeS:string) => {
+                const totalCases = cancer.total;
+                const cancerAlterations:ICancerAlterationPlotData = _.omit(cancer, ['total']);
+                const altTotalCount = _.reduce(cancerAlterations, (total:number, value:number) => total + value, 0);
+                const altTotalPercent = this.toRoundedNumber(altTotalCount / totalCases * 100);
+                if (_.intersection(["all", "", cancerTypeS.toLowerCase()], msv.split(",").map(val => val.toLowerCase())).length && totalCases >= this.totalCasesValue) {
```

Comment:
lets just call msv "multiselectvalue." 
take the msv.split.map statement and assign it to a local variable above that gives reader a sense of what it represents.   like, selected types or something.  just too hard to read.
i think you could also do toLowerCase operation on msv prior to split, to avoid map.
also, why is empty string "" a possibility?

Reply:
The empty string is a possibility because when you deselect all values in the multiselect, it's value is returned back as `""`